### PR TITLE
aws_transit_gateway_multicast_domain

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource. No integration tests with terraform are done as there is no terraform resource available.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_transit_gateway_multicast_domain.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+Ensure that transit gateway connect exists.
+
+    describe aws_transit_gateway_connect(transit_gateway_attachment_ids: 'tgw-attach-1234567890') do
+      it { should exits }
+    end
+
+## Parameters
+
+`transit_gateway_attachment_ids` _(required)_
```
Comment (Nirbhay1997) [inline docs/resources/aws_transit_gateway_multicast_domain.md:22]:
```suggestion
`transit_gateway_attachment_id` _(required)_
```

File: docs/resources/aws_transit_gateway_multicast_domain.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+| options (static_sources_support) | The options for the transit gateway multicast domain. Indicates whether support for statically configuring transit gateway multicast group sources is turned on. |
+| options (auto_accept_shared_associations) | The options for the transit gateway multicast domain. Indicates whether to automatically cross-account subnet associations that are associated with the transit gateway multicast domain. |
+| state | The state of the transit gateway multicast domain. |
+| creation_time | The time the transit gateway multicast domain was created. |
+| tags | The tags for the transit gateway multicast domain. |
+
+## Examples
+
+### Ensure a transit gateway multicast domain id is available.
+    describe aws_transit_gateway_connect(transit_gateway_attachment_ids: 'tgw-attach-1234567890') do
```
Comment (Nirbhay1997) [inline docs/resources/aws_transit_gateway_multicast_domain.md:46]:
```suggestion
    describe aws_transit_gateway_connect(transit_gateway_attachment_id: 'tgw-attach-1234567890') do
```
There are multiple usages we can change that

File: docs/resources/aws_transit_gateway_multicast_domains.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Use the `aws_transit_gateway_multicast_domains` InSpec audit resource to test properties of a plural AWS EC2 TransitGatewayMulticastDomain.
+
+The AWS::EC2::TransitGatewayMulticastDomain resource creates a Connect attachment from a specified transit gateway attachment.
+
+## Syntax
+
+Ensure that transit gateway multicast domain exists.
+
+    describe aws_transit_gateway_multicast_domains(transit_gateway_multicast_domain_ids: "tgw-mcast-domain-1234567890") do
```
Comment (Nirbhay1997) [inline docs/resources/aws_transit_gateway_multicast_domains.md:16]:
```suggestion
    describe aws_transit_gateway_multicast_domains(transit_gateway_multicast_domain_id: "tgw-mcast-domain-1234567890") do
```
same

File: libraries/aws_transit_gateway_multicast_domains.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  attr_reader :table
+
+  FilterTable.create
+             .register_column(:transit_gateway_multicast_domain_ids,            field: :transit_gateway_multicast_domain_id)
+             .register_column(:transit_gateway_ids,                             field: :transit_gateway_id)
+             .register_column(:transit_gateway_multicast_domain_arns,           field: :transit_gateway_multicast_domain_arn)
+             .register_column(:owner_ids,                                       field: :owner_id)
+             .register_column(:options,                                         field: :options)
+             .register_column(:states,                                          field: :state)
+             .register_column(:creation_times,                                  field: :creation_time)
```
Comment (Nirbhay1997) [inline libraries/aws_transit_gateway_multicast_domains.rb:28]:
```suggestion
             .register_column(:creation_time,                                  field: :creation_time)
```
typo

File: test/integration/verify/controls/aws_transit_gateway_multicast_domains.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    it { should exist }
+  end
+
+  describe aws_transit_gateway_multicast_domains do
+    its('transit_gateway_multicast_domain_ids') { should include '' }
+    its('transit_gateway_ids') { should include '' }
+    its('transit_gateway_multicast_domain_arns') { should include '' }
+    its('owner_ids') { should include '' }
+    its('states') { should include 'available' }
+    its('tags') { should_not be_empty }
```
Comment (sathish-progress) [inline test/integration/verify/controls/aws_transit_gateway_multicast_domains.rb:14]: ehh these does not test anything actually ..

File: test/integration/verify/controls/aws_transit_gateway_multicast_domain.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    it { should exist }
+  end
+
+  describe aws_transit_gateway_multicast_domain(transit_gateway_multicast_domain_id: "") do
+    its('transit_gateway_multicast_domain_id') { should eq '' }
+    its('transit_gateway_id') { should eq '' }
+    its('transit_gateway_multicast_domain_arn') { should eq '' }
+    its('owner_id') { should eq '' }
+    its('state') { should eq 'available' }
+    its('tags') { should_not be_empty }
```
Comment (sathish-progress) [inline test/integration/verify/controls/aws_transit_gateway_multicast_domain.rb:14]: this is not testing anything

**Comment by sathish-progress** [comment]
please update readme as well

File: docs/resources/aws_transit_gateway_multicast_domain.md (hunk 1 above)
Comment (soumyo13) [reply docs/resources/aws_transit_gateway_multicast_domain.md:22]: Done

File: docs/resources/aws_transit_gateway_multicast_domain.md (hunk 2 above)
Comment (soumyo13) [reply docs/resources/aws_transit_gateway_multicast_domain.md:46]: Done

File: docs/resources/aws_transit_gateway_multicast_domains.md (hunk 3 above)
Comment (soumyo13) [reply docs/resources/aws_transit_gateway_multicast_domains.md:16]: Done

File: test/integration/verify/controls/aws_transit_gateway_multicast_domain.rb (hunk 6 above)
Comment (soumyo13) [reply test/integration/verify/controls/aws_transit_gateway_multicast_domain.rb:14]: I have skipped it.

File: test/integration/verify/controls/aws_transit_gateway_multicast_domains.rb (hunk 5 above)
Comment (soumyo13) [reply test/integration/verify/controls/aws_transit_gateway_multicast_domains.rb:14]: I have skipped it.

**Comment by soumyo13** [comment]
Done with the changes. Please check.

**Review by sathish-progress — APPROVED** [review]
LGTM

**Pull request merged**